Source/Session.swift

```diff
@@ -859,33 +859,38 @@ open class Session {
     }
 
     func performSetupOperations(for request: Request, convertible: URLRequestConvertible) {
+
+        let initialRequest: URLRequest
+
         do {
-            let initialRequest = try convertible.asURLRequest()
+            initialRequest = try convertible.asURLRequest()
             try initialRequest.validate()
-            rootQueue.async { request.didCreateInitialURLRequest(initialRequest) }
+        } catch {
+            rootQueue.async { request.didFailToCreateURLRequest(with: error.asAFError(default: .createURLRequestFailed(error: error))) }
+            return
+        }
 
-            guard !request.isCancelled else { return }
+        rootQueue.async { request.didCreateInitialURLRequest(initialRequest) }
+
+        guard !request.isCancelled else { return }
 
-            if let adapter = adapter(for: request) {
-                adapter.adapt(initialRequest, for: self) { result in
-                    do {
-                        let adaptedRequest = try result.get()
-                        try adaptedRequest.validate()
-
-                        self.rootQueue.async {
-                            request.didAdaptInitialRequest(initialRequest, to: adaptedRequest)
-                            self.didCreateURLRequest(adaptedRequest, for: request)
-                        }
-                    } catch {
-                        let adaptError = AFError.requestAdaptationFailed(error: error)
-                        self.rootQueue.async { request.didFailToAdaptURLRequest(initialRequest, withError: adaptError) }
-                    }
+        guard let adapter = adapter(for: request) else {
+            rootQueue.async { self.didCreateURLRequest(initialRequest, for: request) }
+            return
+        }
+
+        adapter.adapt(initialRequest, for: self) { result in
+            do {
+                let adaptedRequest = try result.get()
+                try adaptedRequest.validate()
+
+                self.rootQueue.async {
+                    request.didAdaptInitialRequest(initialRequest, to: adaptedRequest)
+                    self.didCreateURLRequest(adaptedRequest, for: request)
                 }
-            } else {
-                rootQueue.async { self.didCreateURLRequest(initialRequest, for: request) }
+            } catch {
+                self.rootQueue.async { request.didFailToAdaptURLRequest(initialRequest, withError: error.asAFError(default: .requestAdaptationFailed(error: error))) }
```

Comment:
We likely always want this wrapped in an `AFError`, so users know its an adapter error.

Reply:
I'm pretty sure we don't need to check whether the error is already `.requestAdaptationFailed`.